test/hotspot/jtreg/compiler/vectorapi/VectorFusedMultiplyAddSubTest.java

```diff
@@ -0,0 +1,410 @@
+            da[i] = RD.nextDouble(25.0);
+            db[i] = RD.nextDouble(25.0);
+            dc[i] = RD.nextDouble(25.0);
+            m[i] = RD.nextBoolean();
+        }
+    }
+
+    interface BTenOp {
+        byte apply(byte a, byte b, byte c);
+    }
+
+    interface STenOp {
+        short apply(short a, short b, short c);
+    }
+
+    interface ITenOp {
+        int apply(int a, int b, int c);
+    }
+
+    interface LTenOp {
+        long apply(long a, long b, long c);
+    }
+
+    interface FTenOp {
+        float apply(float a, float b, float c);
+    }
+
+    interface DTenOp {
+        double apply(double a, double b, double c);
+    }
+
+    private static void assertArrayEquals(byte[] r, byte[] a, byte[] b, byte[] c, boolean[] m, BTenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % B_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    private static void assertArrayEquals(short[] r, short[] a, short[] b, short[] c, boolean[] m, STenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % S_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    private static void assertArrayEquals(int[] r, int[] a, int[] b, int[] c, boolean[] m, ITenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % I_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    private static void assertArrayEquals(long[] r, long[] a, long[] b, long[] c, boolean[] m, LTenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % L_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    private static void assertArrayEquals(float[] r, float[] a, float[] b, float[] c, boolean[] m, FTenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % F_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    private static void assertArrayEquals(double[] r, double[] a, double[] b, double[] c, boolean[] m, DTenOp f) {
+       for (int i = 0; i < LENGTH; i++) {
+           if (m[i % D_SPECIES.length()]) {
+               Asserts.assertEquals(f.apply(a[i], b[i], c[i]), r[i]);
+           } else {
+               Asserts.assertEquals(a[i], r[i]);
+           }
+       }
+    }
+
+    @Test
+    @IR(failOn = { "sve_add", "sve_mul" })
```

Comment:
Do you also want something like `@IR(counts = { "sve_mla", 1 })` here to make sure the fused instruction is generated, in addition to not generating the two un-fused instructions?

Reply:
Yeah, I think `@IR(counts = { "sve_mla", ">=1" })` is better. I will update the tests. Thanks for the review!